<a href="https://colab.research.google.com/github/Monkeyorphan/data_2_checks/blob/main/KnowledgeCheck1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Import the necessary libraries

In [ ]:
import requests
import json
import pandas as pd

Use the get method to request the API and assign it to the variable "API", then check the status code. Status Code 200 means everything is OK and the connection is successful.

In [ ]:
API = requests.get('http://universities.hipolabs.com/search?country=United+Kingdom')
API

<Response [200]>

Pull the data from the API and store it in the "data" variable in the text format

In [ ]:
data = API.text
data

'[{"country": "United Kingdom", "web_pages": ["https://futureworks.ac.uk/"], "name": "Futureworks", "state-province": null, "alpha_two_code": "GB", "domains": ["futureworks.ac.uk"]}, {"country": "United Kingdom", "web_pages": ["https://www.arts.ac.uk"], "name": "University of the Arts London", "state-province": null, "alpha_two_code": "GB", "domains": ["arts.ac.uk"]}, {"country": "United Kingdom", "web_pages": ["http://www.abdn.ac.uk/"], "name": "University of Aberdeen", "state-province": null, "alpha_two_code": "GB", "domains": ["abdn.ac.uk", "aberdeen.ac.uk"]}, {"country": "United Kingdom", "web_pages": ["http://www.aber.ac.uk/"], "name": "University of Wales, Aberystwyth", "state-province": null, "alpha_two_code": "GB", "domains": ["aber.ac.uk"]}, {"country": "United Kingdom", "web_pages": ["http://www.abertay.ac.uk/"], "name": "University of Abertay Dundee", "state-province": null, "alpha_two_code": "GB", "domains": ["abertay.ac.uk"]}, {"country": "United Kingdom", "web_pages": ["h

Convert the text into a json format, parsing it through json using the loads method and then save it to the variable "json"

In [ ]:
json = json.loads(data)

Print the data in json format

In [ ]:
print(json)

[{'country': 'United Kingdom', 'web_pages': ['https://futureworks.ac.uk/'], 'name': 'Futureworks', 'state-province': None, 'alpha_two_code': 'GB', 'domains': ['futureworks.ac.uk']}, {'country': 'United Kingdom', 'web_pages': ['https://www.arts.ac.uk'], 'name': 'University of the Arts London', 'state-province': None, 'alpha_two_code': 'GB', 'domains': ['arts.ac.uk']}, {'country': 'United Kingdom', 'web_pages': ['http://www.abdn.ac.uk/'], 'name': 'University of Aberdeen', 'state-province': None, 'alpha_two_code': 'GB', 'domains': ['abdn.ac.uk', 'aberdeen.ac.uk']}, {'country': 'United Kingdom', 'web_pages': ['http://www.aber.ac.uk/'], 'name': 'University of Wales, Aberystwyth', 'state-province': None, 'alpha_two_code': 'GB', 'domains': ['aber.ac.uk']}, {'country': 'United Kingdom', 'web_pages': ['http://www.abertay.ac.uk/'], 'name': 'University of Abertay Dundee', 'state-province': None, 'alpha_two_code': 'GB', 'domains': ['abertay.ac.uk']}, {'country': 'United Kingdom', 'web_pages': ['ht

Print the entire DataFrame, saving it to the variable "df"

In [ ]:
df = pd.DataFrame(json)
df

country                      web_pages  \
0    United Kingdom   [https://futureworks.ac.uk/]   
1    United Kingdom       [https://www.arts.ac.uk]   
2    United Kingdom       [http://www.abdn.ac.uk/]   
3    United Kingdom       [http://www.aber.ac.uk/]   
4    United Kingdom    [http://www.abertay.ac.uk/]   
..              ...                            ...   
343  United Kingdom  [http://www.wimbledon.ac.uk/]   
344  United Kingdom        [http://www.wlv.ac.uk/]   
345  United Kingdom       [http://www.worc.ac.uk/]   
346  United Kingdom       [http://www.york.ac.uk/]   
347  United Kingdom      [https://www.mbro.ac.uk/]   

                                 name state-province alpha_two_code  \
0                         Futureworks           None             GB   
1       University of the Arts London           None             GB   
2              University of Aberdeen           None             GB   
3    University of Wales, Aberystwyth           None             GB   
4        University of Abertay Dundee           None             GB   
..                                ...            ...            ...   
343           Wimbledon School of Art           None             GB   
344       University of Wolverhampton           None             GB   
345           University of Worcester           None             GB   
346                University of York           None             GB   
347             Middlesbrough College           None             GB   

                            domains  
0               [futureworks.ac.uk]  
1                      [arts.ac.uk]  
2      [abdn.ac.uk, aberdeen.ac.uk]  
3                      [aber.ac.uk]  
4                   [abertay.ac.uk]  
..                              ...  
343               [wimbledon.ac.uk]  
344                     [wlv.ac.uk]  
345                    [worc.ac.uk]  
346                    [york.ac.uk]  
347  [middlesbro.ac.uk, mbro.ac.uk]  

[348 rows x 6 columns]

Take a look at the shape of the DataFrame to see that there are 348 Rows and 6 Columns

In [ ]:
df.shape

(348, 6)

Drop unnecessary columns

In [ ]:
df.drop(columns=['domains', 'state-province', 'alpha_two_code'], inplace=True)

Rename the remaining columns

In [ ]:
df.columns = ['Country', 'Web Page', 'University']
df

Country                       Web Page  \
0    United Kingdom   [https://futureworks.ac.uk/]   
1    United Kingdom       [https://www.arts.ac.uk]   
2    United Kingdom       [http://www.abdn.ac.uk/]   
3    United Kingdom       [http://www.aber.ac.uk/]   
4    United Kingdom    [http://www.abertay.ac.uk/]   
..              ...                            ...   
343  United Kingdom  [http://www.wimbledon.ac.uk/]   
344  United Kingdom        [http://www.wlv.ac.uk/]   
345  United Kingdom       [http://www.worc.ac.uk/]   
346  United Kingdom       [http://www.york.ac.uk/]   
347  United Kingdom      [https://www.mbro.ac.uk/]   

                           University  
0                         Futureworks  
1       University of the Arts London  
2              University of Aberdeen  
3    University of Wales, Aberystwyth  
4        University of Abertay Dundee  
..                                ...  
343           Wimbledon School of Art  
344       University of Wolverhampton  
345           University of Worcester  
346                University of York  
347             Middlesbrough College  

[348 rows x 3 columns]

Filter the DataFrame to show only Universities that contain "London"

In [ ]:
df[df['University'].str.contains('London')]

Country                        Web Page  \
1    United Kingdom        [https://www.arts.ac.uk]   
5    United Kingdom        [http://www.aiuniv.edu/]   
9    United Kingdom           [http://www.aul.edu/]   
13   United Kingdom         [http://www.bbk.ac.uk/]   
21   United Kingdom  [http://www.brijnet.org/lsjs/]   
..              ...                             ...   
324  United Kingdom         [http://www.ucl.ac.uk/]   
328  United Kingdom         [http://www.uel.ac.uk/]   
330  United Kingdom       [http://www.ulsop.ac.uk/]   
332  United Kingdom        [http://www.umds.ac.uk/]   
333  United Kingdom         [http://www.unl.ac.uk/]   

                                            University  
1                        University of the Arts London  
5        American InterContinental University - London  
9                    The American University in London  
13              Birkbeck College, University of London  
21                     London School of Jewish Studies  
..                                                 ...  
324    University College London, University of London  
328                          University of East London  
330           School of Pharmacy, University of London  
332  United Medical and Dental Schools, University ...  
333                         University of North London  

[90 rows x 3 columns]

Select and print the first four rows of the second and third columns of the DataFrame

In [ ]:
df[['Web Page', 'University']][0:4]

Web Page                        University
0  [https://futureworks.ac.uk/]                       Futureworks
1      [https://www.arts.ac.uk]     University of the Arts London
2      [http://www.abdn.ac.uk/]            University of Aberdeen
3      [http://www.aber.ac.uk/]  University of Wales, Aberystwyth